### PR TITLE
Output Fixes

### PR DESCRIPTION
**Pull request by aeslaughter**
This fixes two issues:
1. The Console did nothing with "execute_on = final ...", it now has some minimal outputs including postprocessor/scalar tables. Having all the linear/nonlinear information print on the final timestep is not really possible with the current design, because EXEC_FINAL is a single call to the output function that occurs when the solve is done. @jasondhales I am not sure if you wanted to see the linear/nonlinear iteration information, but implementing this would be quite a large change. Generally, I recommend setting the interval at a lower level (i.e., in a Exodus sub-block).
2. The show/hide listing did not work in certain situations when on a single variable, postprocessor, etc. exists for each type. I fixed it.

### CODE REVIEW

**Comment by jasondhales** [comment]
Of course, it would be helpful to have the iteration information from the
final solve since that is often where users will look.  I still think it is
worthwhile, but this is not a case that we see very often.  I can see
making it a lower priority.

Thanks, Andrew, for looking into this so quickly.

On Mon, Oct 5, 2015 at 2:57 PM, Andrew E Slaughter <notifications@github.com
> wrote:

> This fixes two issues:
>
>    1.
>
>    The Console did nothing with "execute_on = final ...", it now has some
>    minimal outputs including postprocessor/scalar tables. Having all the
>    linear/nonlinear information print on the final timestep is not really
>    possible with the current design, because EXEC_FINAL is a single call to
>    the output function that occurs when the solve is done. @jasondhales
>    <https://github.com/jasondhales> I am not sure if you wanted to see
>    the linear/nonlinear iteration information, but implementing this would be
>    quite a large change. Generally, I recommend setting the interval at a
>    lower level (i.e., in a Exodus sub-block).
>    2.
>
>    The show/hide listing did not work in certain situations when on a
>    single variable, postprocessor, etc. exists for each type. I fixed it.
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/idaholab/moose/pull/5777
> Commit Summary
>
>    - Improved behavior of Console when 'execute_on' includes 'final'
>    - Fixes bug in show/hide
>    - Re-gold test for fix to show/hide list handling
>    - Updated gold for fix to outputs show/hide lists
>
> File Changes
>
>    - *M* framework/include/outputs/AdvancedOutputUtils.h
>    <https://github.com/idaholab/moose/pull/5777/files#diff-0> (24)
>    - *M* framework/src/actions/CommonOutputAction.C
>    <https://github.com/idaholab/moose/pull/5777/files#diff-1> (4)
>    - *M* framework/src/outputs/AdvancedOutput.C
>    <https://github.com/idaholab/moose/pull/5777/files#diff-2> (9)
>    - *M* framework/src/outputs/AdvancedOutputUtils.C
>    <https://github.com/idaholab/moose/pull/5777/files#diff-3> (4)
>    - *M* framework/src/outputs/Console.C
>    <https://github.com/idaholab/moose/pull/5777/files#diff-4> (19)
>    - *M* framework/src/outputs/Output.C
>    <https://github.com/idaholab/moose/pull/5777/files#diff-5> (14)
>    - *M* test/tests/misc/selective_reinit/gold/selective_reinit_test_out.e
>    <https://github.com/idaholab/moose/pull/5777/files#diff-6> (0)
>    - *A* test/tests/outputs/console/console_final.i
>    <https://github.com/idaholab/moose/pull/5777/files#diff-7> (107)
>    - *M* test/tests/outputs/console/tests
>    <https://github.com/idaholab/moose/pull/5777/files#diff-8> (6)
>    - *D* test/tests/outputs/variables/gold/out.e
>    <https://github.com/idaholab/moose/pull/5777/files#diff-9> (0)
>    - *A* test/tests/outputs/variables/gold/output_vars_test_out.e
>    <https://github.com/idaholab/moose/pull/5777/files#diff-10> (0)
>    - *A* test/tests/outputs/variables/gold/show_single_vars_out.e
>    <https://github.com/idaholab/moose/pull/5777/files#diff-11> (0)
>    - *M* test/tests/outputs/variables/output_vars_test.i
>    <https://github.com/idaholab/moose/pull/5777/files#diff-12> (5)
>    - *A* test/tests/outputs/variables/show_single_vars.i
>    <https://github.com/idaholab/moose/pull/5777/files#diff-13> (107)
>    - *M* test/tests/outputs/variables/tests
>    <https://github.com/idaholab/moose/pull/5777/files#diff-14> (9)
>
> Patch Links:
>
>    - https://github.com/idaholab/moose/pull/5777.patch
>    - https://github.com/idaholab/moose/pull/5777.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5777>.
>



-- 
Jason D. Hales, PhD
Department Manager
Fuels Modeling & Simulation
Idaho National Laboratory
(208) 526-2293

**Comment by aeslaughter** [comment]
This is ready, I have a patch waiting for BISON.

**Comment by permcody** [comment]
OK - Reviewing...

File: framework/src/outputs/Console.C [hunk 1]
```diff
@@ -254,6 +260,17 @@ Console::initialSetup()
   if (getParam<bool>("setup_log_early"))
     write(Moose::setup_perf_log.get_perf_info());
 
+  // If the user adds "final or failed" to the execute on, append this to the postprocessors, scalars, etc., but only
+  // if the parameter (e.g., postprocessor_execute_on) has not been modified by the user.
+  if (_execute_on.contains("final"))
```
Comment (permcody) [inline framework/src/outputs/Console.C]: This comment doesn't seem to match this line of code: "final or failed"

File: framework/include/outputs/AdvancedOutputUtils.h [hunk 2]
```diff
@@ -156,6 +156,30 @@ class OutputDataWarehouse : public OutputMapWrapper<OutputData>
...
    */
   OutputDataWarehouse();
+
+  /**
+   * False when the show lists for all variables is empty.
+   *
+   * When false everything should output.
+   * @see AdvancedOutput::initOutputList
+   */
+  bool hasShowList(){ return _has_show_list; }
```
Comment (permcody) [inline framework/include/outputs/AdvancedOutputUtils.h]: Minor comment: Generally I think we put spaces between the parans and braces when doing one liners.

**Pull request merged**